src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3067 @@
+            sub_02014510((NarcId)sp40.narcID, sp40.charDataID, HEAP_ID_REGISTER_HALL_OF_FAME, &sp20[1], hofMon->frontspriteCharbuf[1], hofMon->personality, TRUE, 2, hofMon->species);
+            GetPokemonSpriteCharAndPlttNarcIds(&sp40, pokemon, MON_PIC_FACING_BACK);
+            sub_02014510((NarcId)sp40.narcID, sp40.charDataID, HEAP_ID_REGISTER_HALL_OF_FAME, &sp20[0], hofMon->backspriteCharbuf[0], hofMon->personality, TRUE, 0, hofMon->species);
+            sub_02014510((NarcId)sp40.narcID, sp40.charDataID, HEAP_ID_REGISTER_HALL_OF_FAME, &sp20[1], hofMon->backspriteCharbuf[1], hofMon->personality, TRUE, 0, hofMon->species);
+            sub_02072914(narc, hofMon->unk_001C, hofMon->species, 1);
+            sub_02072914(narc, hofMon->unk_0044, hofMon->species, 0);
+            ++data->numMons;
+        }
+        ReleaseMonLock(pokemon, encry);
+    }
+    NARC_Delete(narc);
+}
+
+static RegisterHallOfFame_MetLocationType RegisterHallOfFame_GetMetLocationType(RegisterHallOfFameData *data, Pokemon *pokemon, PlayerProfile *profile) {
+    BOOL encry = AcquireMonLock(pokemon);
+    RegisterHallOfFame_MetLocationType ret;
+    int version = GetMonData(pokemon, MON_DATA_GAME_VERSION, NULL);
+    if (version == VERSION_SAPPHIRE || version == VERSION_RUBY || version == VERSION_EMERALD) {
+        ret = REGHOF_METLOC_HOENN;
+    } else if (version == VERSION_FIRE_RED || version == VERSION_LEAF_GREEN) {
+        ret = REGHOF_METLOC_KANTO;
+    } else if (version == VERSION_GAMECUBE) {
+        ret = REGHOF_METLOC_ORRE;
+    } else if (version == VERSION_DIAMOND || version == VERSION_PEARL || version == VERSION_PLATINUM) {
+        ret = REGHOF_METLOC_SINNOH;
+    } else if (GetMonData(pokemon, MON_DATA_FATEFUL_ENCOUNTER, NULL)) {
+        ret = REGHOF_METLOC_FATEFUL;
+    } else if (MonIsInGameTradePoke(pokemon, NPC_TRADE_SHUCKIE_SHUCKLE) == TRUE) {
+        ret = REGHOF_METLOC_SHUCKIE;
+    } else if (MonIsInGameTradePoke(pokemon, NPC_TRADE_KENYA_SPEAROW) == TRUE) {
+        ret = REGHOF_METLOC_KENYA;
+    } else if (PlayerProfile_GetTrainerID(profile) != GetMonData(pokemon, MON_DATA_OTID, NULL)) {
+        ret = REGHOF_METLOC_TRADED;
+    } else {
+        PlayerName_FlatToString(profile, data->strbuf1);
+        GetMonData(pokemon, MON_DATA_OT_NAME_2, data->strbuf2);
+        if (String_Compare(data->strbuf1, data->strbuf2)) {
+            ret = REGHOF_METLOC_TRADED;
+        } else if (GetMonData(pokemon, MON_DATA_MET_LOCATION, NULL) >= METLOC_DAY_CARE_COUPLE) {
+            ret = REGHOF_METLOC_FATEFUL;
+        } else if (GetMonData(pokemon, MON_DATA_EGG_MET_MONTH, NULL) == 0) {
+            ret = REGHOF_METLOC_CAUGHT;
+        } else {
+            ret = REGHOF_METLOC_HATCHED;
+        }
+    }
+    ReleaseMonLock(pokemon, encry);
+    return ret;
+}
+
+static int RegisterHallOfFame_GetMmodelBySpeciesFormGender(int species, u8 form, u8 gender) {
+    int ret;
+
+    if (species <= 0 || species > SPECIES_ARCEUS) {
```

Comment:
we really should have a SPECIES_MAX or MAX_SPECIES constant to use in cases like this